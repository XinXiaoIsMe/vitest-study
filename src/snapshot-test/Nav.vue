<template>
  <div>
    {{ title }}
  </div>
</template>

<script lang="ts" setup>
withDefaults(defineProps<{
  title: string;
}>(), {
  title: ''
});
</script>
