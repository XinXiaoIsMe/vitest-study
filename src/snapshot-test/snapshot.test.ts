import { mount } from '@vue/test-utils';
import Nav from './Nav.vue';

it('should generate Nav.vue snapshot', () => {
  const wrapper = mount(Nav, {
    props: {
      title: '1234',
    },
  });
  // 生成快照
  expect(wrapper.html()).toMatchSnapshot();

  // 更新快照方式
  // 1. watch模式下，终端输入u即可
  // 2. 非watch模式，在命令后面添加 --update 参数即可
});

it('should generate Nav.vue inline snapshot', () => {
  const wrapper = mount(Nav, {
    props: {
      title: '1234',
    },
  });
  // 生成行内快照
  expect(wrapper.html()).toMatchInlineSnapshot(`"<div>1234</div>"`);
});

it('should generate Nav.vue snapshot html file', () => {
  const wrapper = mount(Nav, {
    props: {
      title: '1234',
    },
  });
  // 生成快照到Nav.html文件
  expect(wrapper.html()).toMatchFileSnapshot('./Nav.html');
})
