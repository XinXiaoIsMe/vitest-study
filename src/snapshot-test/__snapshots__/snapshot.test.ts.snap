// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should generate Nav.vue snapshot 1`] = `"<div>1234</div>"`;
